### PR TITLE
feat: make getEthereumjsTxDataFromTransaction not trim extra fields

### LINKED ISSUE #7273: [plugin] customTransactionSchema follow-up
Greetings! Thanks again for merging my work in 
- #7227 

Sadly, it's _still_ not enough to allow for extra fields in transactions. Indeed, this function [`getEthereumjsTxDataFromTransaction`](https://github.com/web3/web3.js/blob/4.x/packages/web3-eth/src/utils/prepare_transaction_for_signing.ts#L37-L56) is trimming down fields [despite being validated earlier](https://github.com/web3/web3.js/blob/4.x/packages/web3-eth/src/utils/prepare_transaction_for_signing.ts#L150)

### PR DESCRIPTION
**Pull request by nicolasbrugneaux**
## Description

This PR refactors `getEthereumjsTxDataFromTransaction` to not trim extra fields anymore whilst still aliasing `input` and `gasLimit`

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run lint` with success and extended the tests and types if necessary.
- [x] I ran `npm run test:unit` with success.
- [x] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [x] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [x] I have updated the `CHANGELOG.md` file in the root folder.
- [x] I have linked Issue(s) with this PR in "Linked Issues" menu.

[ Closes #7273 ]

### CODE REVIEW

**Comment by nicolasbrugneaux** [comment]
I also took the opportunity to modify `SomeNewTxTypeTransaction` class to have a custom field so we can kinda e2e tests plugins using custom fields through `customTransactionSchema`

**Review by avkos — APPROVED** [review]

**Review by jdevcs — APPROVED** [review]

File: packages/web3-eth/src/utils/prepare_transaction_for_signing.ts [hunk 1]
```diff
@@ -36,25 +34,23 @@ import { transactionBuilder } from './transaction_builder.js';
...
-	).accessList,
-	maxPriorityFeePerGas: (
-		transaction as FormatType<PopulatedUnsignedEip1559Transaction, typeof ETH_DATA_FORMAT>
-	).maxPriorityFeePerGas,
-	maxFeePerGas: (
-		transaction as FormatType<PopulatedUnsignedEip1559Transaction, typeof ETH_DATA_FORMAT>
-	).maxFeePerGas,
-});
+) => {
+	const txData = { ...transaction };
```
Comment (jdevcs) [inline packages/web3-eth/src/utils/prepare_transaction_for_signing.ts]: This function change will omit some properties entirely if not defined in the input object, while older function always includes those properties as `undefined`, potentially leading to change where its consumer was expecting presence.

Comment (nicolasbrugneaux) [reply packages/web3-eth/src/utils/prepare_transaction_for_signing.ts]: happy to add them back

Comment (nicolasbrugneaux) [reply packages/web3-eth/src/utils/prepare_transaction_for_signing.ts]: done :)

**Comment by jdevcs** [comment]
@nicolasbrugneaux could you pls test dev release https://www.npmjs.com/package/web3/v/4.13.1-dev.cc99825.0?activeTab=versions in your app.

**Comment by nicolasbrugneaux** [comment]
> @nicolasbrugneaux could you pls test dev release https://www.npmjs.com/package/web3/v/4.13.1-dev.cc99825.0?activeTab=versions in your app.

Hello! I was OOO for a couple of days. I'll get right on testing that.

**Comment by nicolasbrugneaux** [comment]
Had to adjust a little bit my dependencies, but it worked wonders, I suppose once they're all released as latest it won't be necessary to play with `resolutions` anymore.

**Pull request merged**